src/main/java/org/springframework/data/couchbase/core/query/QueryCriteria.java

```diff
@@ -126,29 +130,23 @@ public QueryCriteria gte(@Nullable Object o) {
 	}
 
 	public QueryCriteria startingWith(@Nullable Object o) {
-		if (o instanceof QueryCriteria) {
-			QueryCriteria term = where("\"%\"").plus(o);
-			like(term);
-		} else {
-			like(o.toString() + "%");
-		}
+		operator = "STARTING_WITH";
+		value = new Object[] { o };
+		format = "%1$s like (%3$s||\"%%\")";
 		return this;
 	}
 
 	public QueryCriteria plus(@Nullable Object o) {
 		operator = "PLUS";
 		value = new Object[] { o };
-		format = "(%1$s + %3$s)";
+		format = "(%1$s || %3$s)";
```

Comment:
plus is the same as string concat?

Reply:
"|| "is the correct operator to use. But "+" had been working.